binaryIoTypeB/proto/binaryIoTypeB/v1alpha1/binaryIoTypeB.proto

```diff
@@ -0,0 +1,174 @@
+message ConfigurationResponse {
+    oneof type {
+        ConfigurationGetResponse get = 1;
+        ConfigurationSetResponse set = 2;
+        ConfigurationDescribeResponse describe = 3;
+    }
+}
+
+
+// ============= FunctionControl =================
+// Gets the current value of a single binary channel
+message GetSingle {
+    // Specifies the binary input channel (0 - (number of channels - 1))
+    fixed32 channel = 1;
+}
+
+// Gets the values of all binary channels
+message GetAll {
+    // mask to define which channels are affected by the get all command. 0 means not set, 1 means set, LSB is Output0
+    fixed32 mask = 1;
+}
+
+// FunctionControlGet to pass to Functionblock.FunctionControl.functionSpecificFunctionControlGet hook
+message FunctionControlGet {
+    oneof type {
+        GetSingle single = 1;
+        GetAll all = 2;
+    }
+}
+
+// Sets the value of a single binary channel
+message SetSingle {
+    // Number of the binary output channel (0 - (number of channels - 1))
+    fixed32 channel = 1;
+    // State of the binary output channel. false means low, true means high.
+    bool state = 2;
+}
+
+// Sets all binary channels
+message SetAll {
+    // binary coded map of input values. 0 means low, 1 means high, LSB is Output0
+    fixed32 values = 1;
+    // mask to define which channels are affected by the set all command. 0 means not set, 1 means set, LSB is Output0
+    fixed32 mask = 2;
+}
+
+// FunctionControlSet to pass to Functionblock.FunctionControl.functionSpecificFunctionControlSet hook
+message FunctionControlSet {
+    oneof type {
+        SetSingle single = 1;
+        SetAll all = 2;
+    }
+}
+
+enum Error {
+    // At least one channel to process is configured for the wrong direction and no input was read resp. no output was written
+    BINARYIOTYPEB_WRONG_DIRECTION = 0;
+    // Channel to read/write does not exist
+    BINARYIOTYPEB_INVALID_CHANNEL = 1;
+}
+
+message GetSingleResponse {
+    // Number of the binary input channel
+    fixed32 channel = 1;
+    // State for the bianry input channel
+    bool state = 2;
+}
+
+message GetAllResponse {
+    // binary coded map of input values. 0 means low, 1 means high, LSB is Input0
+    fixed32 inputs = 1;
+}
+
+message GetError {
+    Error error = 1;
+}
+
+// FunctionControlGetResponse to pass to Functionblock.FunctionControlResponse.functionSpecificControlGet hook
+message FunctionControlGetResponse {
+    oneof type {
+        GetSingleResponse single = 1;
+        GetAllResponse all = 2;
+        GetError error = 3;
+    }
+}
+
+message SetSingleResponse {
+    // no parameters
+}
+
+message SetAllResponse {
+    // no parameters
+}
+
+message SetError {
+    Error error = 1;
+}
+
+// FunctionControlSetResponse to pass to Functionblock.FunctionControlResponse.functionSpecificControlSet hook
+message FunctionControlSetResponse {
+    oneof type {
+        SetSingleResponse single = 1;
+        SetAllResponse all = 2;
+        SetError error = 3;
```

Comment:
This is strange. Why not use function block generic errors? BinaryIoTypeA doesn't need such errors

Reply:
There is no generic error that tells the host that the gpio is input/output only.